NavigateU/NavigateU/DocumentRemote/DocumentLocalDataSource.swift

```diff
@@ -0,0 +1,212 @@
+//
+//  DocumentLocalDataSource.swift
+//  NavigateU
+//
+//  Created by Faki Doosuur Doris on 28.05.2024.
+//
+
+import Foundation
+import FirebaseAuth
+import CoreData
+import Combine
+
+protocol DocumentLocalDataSourceProtocol {
+    func likeArticle(article: Article, completion: @escaping (Result<Void, Error>) -> Void)
+    func unlikeArticle(article: Article, completion: @escaping (Result<Void, Error>) -> Void)
+    func getLikedArticles() -> [Article]
+    func isArticleLiked(articleId: Int) -> Bool
+    func getArticle(articleId: Int) -> Article?
+    func removeDuplicateArticles()
+
+    var articleLiked: Bool { get set }
+    var articleUnliked: Bool { get set }
+    var articleLikedPublisher: Published<Bool>.Publisher { get }
+    var articleUnlikedPublisher: Published<Bool>.Publisher { get }
+}
+
+class DocumentLocalDataSource: DocumentLocalDataSourceProtocol {
+    static let shared = DocumentLocalDataSource()
+    @Published var articleLiked: Bool = false
+    @Published var articleUnliked: Bool = false
```

Comment:
looks like this property cannot be equal (you better use one property isLiked instead of two properties)